ttnn/cpp/ttnn/operations/pool/generic/generic_pools.cpp

```diff
@@ -45,109 +45,134 @@ Tensor Pool2DOp<pool_type>::invoke(
     const std::optional<const MemoryConfig>& memory_config,
     const std::optional<const TensorMemoryLayout> applied_shard_scheme,
     bool ceil_mode) {
-    sliding_window::SlidingWindowConfig sliding_window_config{
-            .batch_size = batch_size,
-            .input_hw = {input_h, input_w},
-            .window_hw = {kernel_size.at(0), kernel_size.at(1)},
-            .stride_hw = {stride.at(0), stride.at(1)},
-            .pad_hw = {padding.at(0), padding.at(1)},
-            .dilation_hw = {dilation.at(0), dilation.at(1)},
-            .ceil_mode = ceil_mode,
+
+    Conv2dConfig conv_config = Conv2dConfig();
+    const uint32_t output_height =
+        ((input_h - kernel_size[0] - ((kernel_size[0] - 1) * (dilation[0] - 1)) + 2 * padding[0]) / stride[0]) + 1;
+    const uint32_t output_width =
+        ((input_w - kernel_size[1] - ((kernel_size[0] - 1) * (dilation[0] - 1)) + 2 * padding[1]) / stride[1]) + 1;
+
+    DeviceComputeKernelConfig compute_config = conv::get_conv_default_compute_kernel_config(input_tensor.device());
+
+    const auto compute_grid_size = input_tensor.device()->compute_with_storage_grid_size();
+
+    bool auto_shard = false;
+    if (!input_tensor.is_sharded() && !conv_config.shard_layout.has_value()) {
+        // In this case we deduce the shard layout.
+        // Note: function calculate_L1_usage called by function below
+        //       takes weight tensor estimation into consideration,
+        //       which should be omitted in pool op.
+        conv_config = conv::determine_conv_config_for_auto_shard(
```

Comment:
This function will do the work for conv2d as it knows how to estimate L1 memory usage for conv2d but I don't see how it can do the same for max_pool?